front/src/store/tv/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,77 @@
+import { tvReducer } from '../reducers';
+import {
+  REQUEST_BASIC_SHOW_INFO_SUCCEEDED,
+  SAVE_EPISODE_DATA,
+  SAVE_SEARCH_QUERY,
+  SET_CALENDAR_EPISODES,
+  SET_POPULAR_SHOWS,
+} from '../actions';
+
+describe('tv reducer', () => {
+  //this state is used for SAVE_SEARCH_QUERY tests
+  const commonState = {
+    savedQueries: [{ query: '1' }],
+    episodeData: {},
+    basicShowInfo: {},
+    calendarEpisodesForDisplay: [],
+    popularShows: [],
+  };
+
+  it('returns expected state on SAVE_SEARCH_QUERY', () => {
+    const action = { type: SAVE_SEARCH_QUERY, payload: { query: '1' } };
+    expect(tvReducer(undefined, action)).toEqual(commonState);
+  });
+
+  it('filters existing query on SAVE_SEARCH_QUERY', () => {
+    const action = { type: SAVE_SEARCH_QUERY, payload: { query: '1' } };
+    expect(tvReducer(commonState as any, action)).toEqual(commonState);
+  });
+
+  it('returns expected state on SAVE_EPISODE_DATA', () => {
+    const expectedState = {
+      savedQueries: [],
+      episodeData: { episode: 'test' },
+      basicShowInfo: {},
+      popularShows: [],
+      calendarEpisodesForDisplay: [],
+    };
```

Comment:
I think we can avoid writing the whole state in each block by renaming commonState at the top to blankState and then in each test, like here, do:

```
const expectedState = { ...blankState, episodeData: { episode: 'test' } }
```

Reply:
@trybick  A great suggestion!